pkg/workflows/management/upgrade_management_components.go

```diff
@@ -0,0 +1,163 @@
+package management
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/executables"
+	"github.com/aws/eks-anywhere/pkg/filewriter"
+	"github.com/aws/eks-anywhere/pkg/logger"
+	"github.com/aws/eks-anywhere/pkg/providers"
+	"github.com/aws/eks-anywhere/pkg/task"
+	"github.com/aws/eks-anywhere/pkg/types"
+	"github.com/aws/eks-anywhere/pkg/validations"
+	"github.com/aws/eks-anywhere/pkg/workflows"
+	"github.com/aws/eks-anywhere/pkg/workflows/interfaces"
+)
+
+// UpgradeManagementComponentsRunner is a schema for upgrade management components.
+type UpgradeManagementComponentsRunner struct {
+	provider        providers.Provider
+	clusterManager  interfaces.ClusterManager
+	gitOpsManager   interfaces.GitOpsManager
+	writer          filewriter.FileWriter
+	capiManager     interfaces.CAPIManager
+	eksdInstaller   interfaces.EksdInstaller
+	eksdUpgrader    interfaces.EksdUpgrader
+	clusterUpgrader interfaces.ClusterUpgrader
```

Comment:
do we need all these things? for example, `ClusterUpgrader` doesn't seem to fit here.

Reply:
removing them from here also allows us to not have to build them in the factory

Reply:
Yes, I tried removing each dependency and any missing would cause error.

Reply:
I did another check:
clusterUpgrade is removed
eksdinstaller is kept because it run InstallEksdManifest in task "install-new-eksa-version-components"
writer is kept because it writes "checkpoint" file when the task run failed